src/chronify/time_series_checker.py

```diff
@@ -74,6 +72,40 @@ def _run_timestamp_checks_on_tmp_table(self, table_name: str) -> None:
         result3 = self._conn.execute(text(query3)).fetchone()
         assert result3 is not None
         actual_count = result3[0]
-        if actual_count != self._schema.time_config.length:
-            msg = f"Time arrays must have length={self._schema.time_config.length}. Actual = {actual_count}"
+        expected_count = len(self._time_generator.list_timestamps())
+        if actual_count != expected_count:
+            msg = f"Time arrays must have length={expected_count}. Actual = {actual_count}"
             raise InvalidTable(msg)
+
+
+def compare_lists(actual: list, expected: list) -> None:
```

Comment:
If this is a public function used in other modules, perhaps it needs a better name. check_timestamp_lists?